### DISCUSSION TITLE
virtual:uno.css can not be resolved

### DISCUSSION THREAD

**Discussion by lovetingyuan** [tailwindlabs/tailwindcss#12574] [Help]
[vite] Pre-transform error: Failed to resolve import "virtual:uno.css" from "index.html?html-proxy&index=0.js". Does the file exist?

**Comment by wongjn** [comment]
Could I double-check that this is to do with Tailwind please? Your title and the error seems to mention [`uno.css`](https://unocss.dev/). If it *is* to do with Tailwind with coincidental naming, please consider providing more debugging details, such as a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by lovetingyuan** [reply]
Sorry, I use  `mode: 'shadow-dom' `, this mode should remove `import "virtual:uno.css"`.
But I have a new question, each custom element will be injected `preflights` layer css. It is too redundant and results in this code (layer preflights) being included in the code of every web component when it is finally built.
